src/ArrayContainer.php

```diff
@@ -32,6 +38,7 @@ public function __construct($data = array())
         } else {
             throw new \InvalidArgumentException('The ArrayContainer requires either an array or an array-like object');
         }
+        $this->delegateLookupContainer = $delegateLookupContainer ?: $this;
```

Comment:
Maybe it would be better to do:
```php
$this->delegateContainer = $delegateContainer ? new FailoverOnMissContainer($delegateContainer, $this) : $this;
```

This way dependencies in the original container aren't missed.